### PR TITLE
Add tests for multiple root key rotation 

### PR DESCRIPTION
**Pull request by lukpueh**
**Fixes issue #**:
Add tests for bugfix provided by #885.

**Description of the changes being introduced by the pull request**:
- Add two ed25519 key pairs for updater root key rotation tests
- Add two root key rotation tests

See commit messages for more details.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by lukpueh** [comment]
@trishankatdatadog, would you mind giving me a LGTM on these tests that I added for the bug fix you provided in #885?

File: tests/test_updater_root_rotation_integration.py [hunk 1]
```diff
@@ -213,9 +215,98 @@ def test_root_rotation(self):
...
+    self.assertTrue(filecmp.cmp(
+      os.path.join(self.repository_directory, 'metadata', '4.root.json'),
+      os.path.join(self.repository_directory, 'metadata', 'root.json')))
+
+    # Lower max root rotation cap so that client stops updating early
+    max_rotation_backup = tuf.settings.MAX_NUMBER_ROOT_ROTATIONS
+    tuf.settings.MAX_NUMBER_ROOT_ROTATIONS = 2
+
+    # Update on client 1.root.json --> 2.root.json --> 3.root.json,
+    # but not stop before updating to 4.root.json
```
Comment (trishankatdatadog) [inline tests/test_updater_root_rotation_integration.py]:
```suggestion
    # but stop before updating to 4.root.json
```

File: tests/test_updater_root_rotation_integration.py [hunk 2]
```diff
@@ -213,9 +215,98 @@ def test_root_rotation(self):
...
+    # Assert that repo indeed has "4.root.json" and that it's the latest root
+    self.assertTrue(filecmp.cmp(
+      os.path.join(self.repository_directory, 'metadata', '4.root.json'),
+      os.path.join(self.repository_directory, 'metadata', 'root.json')))
+
+    # Lower max root rotation cap so that client stops updating early
+    max_rotation_backup = tuf.settings.MAX_NUMBER_ROOT_ROTATIONS
+    tuf.settings.MAX_NUMBER_ROOT_ROTATIONS = 2
+
+    # Update on client 1.root.json --> 2.root.json --> 3.root.json,
```
Comment (trishankatdatadog) [inline tests/test_updater_root_rotation_integration.py:297]:
```suggestion
    # Update client from 1.root.json --> 2.root.json --> 3.root.json,
```

**Review by trishankatdatadog — APPROVED** [review]
LGTM, @lukpueh, thanks very much!

Have you tested this PR on both my branch (should ✅), and `master` (should ❌)?

**Comment by lukpueh** [comment]
> Have you tested this PR on both my branch (should ✅), and master (should ❌)?

Yes I did, and yes it did! :) 

Thanks for catching the comment mistake. Will merge once the tests ran through.

**Pull request merged**